packages/flutter_tools/lib/src/ios/xcresult.dart

```diff
@@ -361,7 +428,38 @@ bool _shouldDiscardIssue({
   return false;
 }
 
-List<XCResultIssue> _parseIssuesFromIssueSummariesJson({
+/// [NEW] Helper to parse issues from the new (Xcode 16+) flat list format.
+List<XCResultIssue> _parseIssuesFromNewFormat({
+  required XCResultIssueType type,
+  required Object? jsonList,
+  required List<XCResultIssueDiscarder> issueDiscarders,
+}) {
+  final issues = <XCResultIssue>[];
+  if (jsonList is! List<Object?>) {
+    return issues;
+  }
+
+  for (final Object? issueJson in jsonList) {
+    if (issueJson is! Map<String, Object?>) {
+      continue;
+    }
+    final resultIssue = XCResultIssue.fromNewFormat(type: type, issueJson: issueJson);
+    var discard = false;
+    for (final discarder in issueDiscarders) {
+      if (_shouldDiscardIssue(issue: resultIssue, discarder: discarder)) {
+        discard = true;
+        break;
+      }
+    }
+    if (!discard) {
+      issues.add(resultIssue);
+    }
+  }
+  return issues;
+}
+
+/// [OLD] Renamed helper to parse issues from the old (pre-Xcode 16) format.
+List<XCResultIssue> _parseIssuesFromOldFormat({
```

Comment:
"Old format" probably isn't a good name, as they may add a new format next release, then this would be "old old" 🙂 . How about _parseIssuesFromXcode15Format

Reply:
Done.